### ISSUE TITLE
Wagtail items in menu with defined names instead autogenerated

### ISSUE THREAD

**Issue by Quadric** [wagtail/wagtail#6111]
### Is your proposal related to a problem?

Well i have a problem with menu items in Wagtail. In my installation i am doing many things with menu and menu items - for ex. i need to remove some default items or move them to another menu.

Please take a look at this eample when i need to remove Snippets from "main menu". I just don't want to recreated the whole menu - just remove one item.

```python
@hooks.register('construct_main_menu')
def hide_snippets_menu_item(request, menu_items):
    menu_items[:] = [item for item in menu_items if item.__class__.__name__ != 'SnippetsMenuItem']
```
OK... Why i did that? There is a "name" property for "item" menu elements... but some of them are defined. like for images:

```python
@hooks.register('register_admin_menu_item')
def register_images_menu_item():
    return ImagesMenuItem(
        _('Images'), reverse('wagtailimages:index'),
        name='images', icon_name='image', order=300
    )
```
I hope you see param "name" with "images' value.

But for "Snippets" there is no defined name

```python
@hooks.register('register_admin_menu_item')
def register_snippets_menu_item():
    return SnippetsMenuItem(
        _('Snippets'),
        reverse('wagtailsnippets:index'),
        icon_name='snippet',
        order=500
    )
```

so "name" is automatically generated from `_('Snippets'),` and lower-cased.... so this "menu item" name will be "snippets" - great? No - because it will be named "snippets" if user has "default" or "english" language preference. Well for polish this "menu item" will be called "snippety".

Back again

So if i would like to remove "Images" from menu it's easy as

```python
@hooks.register('construct_main_menu')
def hide_images_menu_item(request, menu_items):
    menu_items[:] = [item for item in menu_items if item.name != 'images']
```

because "ImagesMenuItem" has defined "name". But for "Snippets" i do not know what name will be.

### What i've tried?

Yes i could "Compare" generated "menu item" name with "auto generation" hack like so:

```python
@hooks.register('construct_main_menu')
def hide_snippets_menu_item(request, menu_items):
    menu_items[:] = [item for item in menu_items if item.name != _('Snippets').lower()]
```

but i doesn't work either. Why? Because hook `register_admin_menu_item` is run only once when menu is generated but hook `construct_main_menu` is run on every request - so "menu item" names are generated once... and they can't be compared in function which run's on every request because "language" may by changed.

### Describe the solution you'd like

I believe the easiest non-breaking solution (at least for someone who does not made any extreme advanced changes in menu system) is to add "name" to every existing "MenuItem" registered in Wagtail System. For the "main menu" i think only "Snippets" are lack of "defined" name. But the same problem is for "`register_settings_menu_item`" which i think neither has defined menu name.

Does it make sense? If so i could prepare a Merge Request.

**Comment by gasman** [comment]
Hi @Quadric - thanks for the report. Yes, I'd be happy to accept a pull request that consistently adds `name` attributes to the built-in menu items.

**Comment by gasman** [comment]
Fixed in #10270
